.github/workflows/build.yml

```diff
@@ -113,7 +113,7 @@ jobs:
     needs: checkout-macos
     with:
       build-runs-on: macos-14-xlarge
-      test-runs-on: macos-13
+      test-runs-on: macos-12
```

Comment:
Do we intentionally want to use an x64 runner here? For context, the `macos-12` and `macos-13` images run on x64, and the `macos-13-xlarge` and `macos-14-xlarge` images run arm64: https://github.com/actions/runner-images?tab=readme-ov-file#available-images

Reply:
@MarshallOfSound can confirm but i think we did this to avoid running on more expensive larger machines?

Reply:
@VerteDinde yes - this is to test the x64 build, so we want an x64 runner.  @codebytere pointed out that according to https://github.com/actions/runner-images/issues/9255 `macos-12` is going to be deprecated in Sept and removed by end of year, so I am going to revert this change and stick with `macos-13`.